### PR TITLE
enable ros2 for win10 iot core

### PR DESCRIPTION
**Pull request by bfjelds**
Most of these changes were made while trying to enable filters and laser_filters (using Node namespace, moving from boost to std, etc).  

NO_ERROR is already defined for Windows, so I changed it here to NO_ERRORS.

I wasn't sure how best to handle time in ROS2, tried to get it right :).

### CODE REVIEW

File: tf2/include/tf2/exceptions.h [hunk 1]
```diff
@@ -39,15 +39,18 @@
 
 namespace tf2{
 
+#ifndef TF2ERROR_ENUM
+#define TF2ERROR_ENUM
 enum class TF2Error : std::uint8_t {
-  NO_ERROR = 0,
+  NO_ERRORS = 0,
```
Comment (mikaelarguedas) [inline tf2/include/tf2/exceptions.h:45]: Not sure if this is a requirement but from my understanding, this should match the enums defined in the corresponding TF2Error ROS message https://github.com/ros2/geometry2/blob/fe1858c8155d94d52f3591529379d29192754cb0/tf2_msgs/msg/TF2Error.msg#L1.

For future references: This seems related to existing issues about windows defines: https://github.com/ros/geometry2/issues/172 and https://github.com/ros2/rosidl/issues/118.

Comment (bfjelds) [reply tf2/include/tf2/exceptions.h:45]: Thanks!  I was blissfully unaware of that.  Have pushed a fix to match the message and the enum.

Comment (mikaelarguedas) [reply tf2/include/tf2/exceptions.h:45]: thanks! I'll leave it up to @tfoote to review the rest of this PR and comment on the best way to move forward regarding the enum names as he is the one that has the most context.

**Comment by cottsay** [comment]
Hi @bfjelds, it looks like ther has been progress on some of the referenced PRs. Is this change still relevant? If so, what are the next steps?

**Comment by cottsay** [comment]
Since this change doesn't appear to have much activity recently, I'm going to move it to the backlog for now. @bfjelds, if you find time to revisit this, please reach out.

**Comment by clalancette** [comment]
There hasn't been activity here in quite a while, and this now needs a rebase, so I'm going to close this out.  Feel free to reopen or open a new PR if you want to continue.  Thanks for the contribution.

**Pull request closed without merging**